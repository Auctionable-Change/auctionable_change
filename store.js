import React, { createContext, useContext, useReducer } from "react";

const StoreContext = createContext();

const initialState = {
  currentListing: {
    id: null,
    title: "",
    description: "",
    price: null,
    donor: "",
    donor_email: "",
    status: "",
    category: "",
    charity: "",
    charity_url: "",
    charity_score: "",
    charity_score_image: "",
    image: "",
    bids: [],
  },
  buyerDetails: {
    item_id: null,
    bidder_name: "",
    bidder_email: "",
    amount: null,
    street_address: "",
    city: "",
    state: "",
    zip_code: "",
    receipt: "",
  },
  listingToPost: {},
};

const reducer = (state, action) => {
  switch (action.type) {
    case "ADD_CURRENT_LISTING":
      return {
        ...state,
        currentListing: action.currentListing,
      };
    case "ADD_SELECTED_CHARITY":
      return {
        ...state,
        listingToPost: {
          ...state.listingToPost,
          charity: action.selectedCharity.name,
          charity_url: action.selectedCharity.url,
          charity_score: action.selectedCharity.rating,
          charity_score_image: action.selectedCharity.rating_image,
        },
      };
    case "ADD_BUYER_DETAILS":
      return {
        ...state,
        buyerDetails: { ...state.buyerDetails, ...action.buyerDetails },
      };
    case "ADD_TO_LISTING":
      return {
        ...state,
        listingToPost: { ...state.listingToPost, ...action.listingToPost },
      };
    case "POST_BID":
      return {
        ...state,
        currentBid: action.bid
      };
      case "SET_USER":
      return {
        ...state,
        loginDetails: action.loginDetails,
        userInfo: action.userInfo,
        registrationDetails: action.registration
      }

    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
};

export const StoreProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  return (
    <StoreContext.Provider value={{ state, dispatch }}>
      {children}
    </StoreContext.Provider>
  );
};

export const useStore = () => useContext(StoreContext);
