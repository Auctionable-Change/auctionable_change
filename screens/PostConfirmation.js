import * as React from "react";
import { SafeAreaView, Text } from "react-native";
import { StyleSheet } from "react-native";
import { TouchableOpacity, ScrollView } from "react-native-gesture-handler";
import { useStore } from "../store";

const PostConfirmation = ({ navigation }) => {
  const { state } = useStore()
  console.log(state)
  return (
    <SafeAreaView style={styles.container}>
      <ScrollView>
        <Text>{state.selectedCharity.name}</Text>
        <TouchableOpacity
          onPress={() => navigation.navigate("Confirmation")}
          style={styles.button}
        >
          <Text>View Listing</Text>
        </TouchableOpacity>
      </ScrollView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  button: {
    backgroundColor: "#FFFFFF",
    borderRadius: 10,
    width: 200,
    padding: 10,
    margin: 10,
    alignItems: "center",
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.18,
    shadowRadius: 1.0,
  },
});

export default PostConfirmation;
