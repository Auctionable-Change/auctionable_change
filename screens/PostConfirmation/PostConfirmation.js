import * as React from "react";
import { Image, StyleSheet, View } from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";
import { ScrollView } from "react-native-gesture-handler";
import { useStore } from "../../store";
import { Button, Text } from "native-base";
import { submitListing } from "../apiCalls";

const PostConfirmation = ({ navigation }) => {
  const { state, dispatch } = useStore();

  const handleSubmit = async () => {
     dispatch({
       type: "ADD_TO_LISTING",
       listingToPost: { auction_length: 10080 },
     });
    await submitListing(state.listingToPost);
    navigation.navigate("Thank You");
  };

  return (
    <SafeAreaView style={styles.container}>
      <ScrollView style={{ width: "100%" }}>
        <Text
          style={{
            marginBottom: 20,
            fontSize: 25,
            textAlign: "center",
            fontFamily: "quicksand-bold",
          }}
        >
          Please review your listing before confirming:
        </Text>
        <Image
          source={{ uri: state.listingToPost.image }}
          style={styles.image}
        />
        <View style={{ padding: 10, backgroundColor: "#FFFFFF" }}>
          <Text
            style={{ fontFamily: "quicksand-bold", fontSize: 18, marginTop: 5 }}
          >
            Details:
          </Text>
          <Text style={styles.text}>Title: {state.listingToPost.title}</Text>
          <Text style={styles.text}>Price: ${state.listingToPost.price}</Text>
          <Text style={styles.text}>
            Description: {state.listingToPost.description}
          </Text>
          <Text style={styles.text}>
            Category: {state.listingToPost.category}
          </Text>
          <Text style={styles.text}>
            Charity: {state.listingToPost.charity}
          </Text>
        </View>
        <Text style={styles.alert}>
          Navigate back to the previous screens to edit or confirm listing.
        </Text>
      </ScrollView>
      <Button
        block
        success
        onPress={() => handleSubmit()}
        style={{ marginTop: 10, backgroundColor: "#2cb833" }}
      >
        <Text style={{ fontSize: 15, fontFamily: "quicksand-bold" }}>
          Confirm Listing
        </Text>
      </Button>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  cardItem: {
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.18,
    shadowRadius: 1.0,
    borderRadius: 3,
  },
  image: {
    width: "100%",
    height: 400,
    resizeMode: "cover",
  },
  text: {
    marginBottom: 5,
    marginLeft: 10,
    fontFamily: "quicksand",
    fontSize: 15,
  },
  alert: {
    textAlign: "center",
    margin: 10,
    width: "80%",
    alignSelf: "center",
    color: "grey",
    fontFamily: "quicksand",
    fontSize: 15,
  },
});

export default PostConfirmation;
