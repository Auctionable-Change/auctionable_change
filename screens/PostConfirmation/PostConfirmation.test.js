import React from 'react'
import { render, fireEvent } from 'react-native-testing-library';
import PostConfirmation from './PostConfirmation';
import { StoreProvider } from '../../store'
import {expect, it} from '@jest/globals';
import renderer from "react-test-renderer";


describe('PostConfirmation', () => {
  it('should render', () => {
    const tree = renderer.create(
      <StoreProvider>
        <PostConfirmation />
      </StoreProvider>).toJSON();
   expect(tree).toMatchSnapshot();
  })

})