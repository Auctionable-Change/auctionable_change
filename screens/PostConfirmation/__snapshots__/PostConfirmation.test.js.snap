// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostConfirmation should render 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RCTScrollView
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "fontFamily": "quicksand-bold",
              "fontSize": 25,
              "marginBottom": 20,
              "textAlign": "center",
            },
          ]
        }
      >
        Please review your listing before confirming:
      </Text>
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "height": 400,
            "resizeMode": "cover",
            "width": "100%",
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "fontFamily": "quicksand-bold",
                "fontSize": 18,
                "marginTop": 5,
              },
            ]
          }
        >
          Details:
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "fontFamily": "quicksand",
                "fontSize": 15,
                "marginBottom": 5,
                "marginLeft": 10,
              },
            ]
          }
        >
          Title: 
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "fontFamily": "quicksand",
                "fontSize": 15,
                "marginBottom": 5,
                "marginLeft": 10,
              },
            ]
          }
        >
          Price: $
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "fontFamily": "quicksand",
                "fontSize": 15,
                "marginBottom": 5,
                "marginLeft": 10,
              },
            ]
          }
        >
          Description: 
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "fontFamily": "quicksand",
                "fontSize": 15,
                "marginBottom": 5,
                "marginLeft": 10,
              },
            ]
          }
        >
          Category: 
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "fontFamily": "quicksand",
                "fontSize": 15,
                "marginBottom": 5,
                "marginLeft": 10,
              },
            ]
          }
        >
          Charity: 
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "alignSelf": "center",
              "color": "grey",
              "fontFamily": "quicksand",
              "fontSize": 15,
              "margin": 10,
              "textAlign": "center",
              "width": "80%",
            },
          ]
        }
      >
        Navigate back to the previous screens to edit or confirm listing.
      </Text>
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#2cb833",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 5,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 2,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "marginTop": 10,
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 16.5,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          },
          Object {
            "fontFamily": "quicksand-bold",
            "fontSize": 15,
          },
        ]
      }
    >
      Confirm Listing
    </Text>
  </View>
</RNCSafeAreaView>
`;
